### PR TITLE
DCMI Type Enrichments

### PR DESCRIPTION
**Pull request by no-reply**
Adds two enrichments:

  - A DCMI Type mapper, using a hash map and White Similarity to match strings to DCMI Types; and
  - A DCMI Type enforcer to unceremoniously destroy any non DCMI values, or invalid URIs.

### CODE REVIEW

**Comment by anarchivist** [comment]
Does this supersede #146?

**Comment by no-reply** [comment]
Yes. I closed #146 before working on this.

**Comment by no-reply** [comment]
This should be in for ESDN enrichment.

File: lib/krikri/enrichments/dcmi_type_map.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    'costume' => RDF::DCMITYPE.Image,
+                    'statue' => RDF::DCMITYPE.Image,
+                    'sculpture' => RDF::DCMITYPE.Image,
+                    'container' => RDF::DCMITYPE.Image,
+                    'jewelry' => RDF::DCMITYPE.Image,
+                    'furnishing' => RDF::DCMITYPE.Image,
+                    'furniture' => RDF::DCMITYPE.Image,
+                    'drawing' => RDF::DCMITYPE.Image,
+                    'print' => RDF::DCMITYPE.Image,
+                    'paniting' => RDF::DCMITYPE.Image,
```
Comment (anarchivist) [inline lib/krikri/enrichments/dcmi_type_map.rb]: typo: should be `'painting'`

Comment (no-reply) [inline lib/krikri/enrichments/dcmi_type_map.rb]: nice

File: lib/krikri/enrichments/dcmi_type_map.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    private
+
+    ##
+    # Performs White Similarity comparison against the keys, and gives the
+    # value of the closest match.
+    #
+    # @param value [String] a string value to compare to the hash map keys.
+    # @param threshold [Float] the value at which a string is considered to
+    #   be a match
+    #
```
Comment (anarchivist) [inline lib/krikri/enrichments/dcmi_type_map.rb]: maybe a `@see` reference to `Text::WhiteSimilarity` here?

Comment (no-reply) [inline lib/krikri/enrichments/dcmi_type_map.rb]: Yeah, and probably a link to a resource describing the algorithm and a `@todo` to evaluate text comparison methodologies/move text comparison logic to a utility.

:+1:

Comment (anarchivist) [inline lib/krikri/enrichments/dcmi_type_map.rb]: Yes, that was going to be my next suggestion. I was skimming the original article, trying to understand the similarity threshold as we're using it here.

**Comment by anarchivist** [comment]
Other than addressing typos and comments, and squashing, I'm :+1:

**Comment by no-reply** [comment]
squashed!

**Pull request merged**